File: Python3/Minimum Add to Make Parentheses Valid.py
# 32ms, 32.71%; 12.7MB, 100.00%
class Solution:
    def minAddToMakeValid(self, S: str) -> int:
        st = []
        for c in S:
            if st and st[-1] == '(' and c == ')':
                st.pop()
            else:
                st.append(c)
        return len(st)

File: Java/Minimum Add to Make Parentheses Valid.java
// 2ms, 34.21%; 34.5MB, 100.00%
class Solution {
    public int minAddToMakeValid(String S) {
        Stack<Character> st = new Stack<Character>();
        for (int i = 0, size = S.length(); i < size; ++i) {
            if (!st.empty() && st.peek() == '(' && S.charAt(i) == ')') {
                st.pop();
            } else {
                st.push(S.charAt(i));
            }
        }
        return st.size();
    }
}
